lib/stat_tracker.rb

```diff
@@ -80,13 +94,50 @@ def team_info
   end
 
 #---------------------------
-  private
+  # private
 
-  def coach_list_wins_losses
-    coach_hash = Hash.new
-    game_teams.each do |gt|
-      (coach_hash[gt.head_coach] ||= []) << gt.result
+def game_team_results_by_season(season)
+    games_of_season = games.find_all do |game|
+      game['season'] == season
+    end
+    game_ids_in_season = games_of_season.map do |game|
+      game['game_id']
+    end
+    @games_results_per_season = game_teams.find_all do |gt|
+      game_ids_in_season.include? gt['game_id']
     end
-    coach_hash
-  end  
+  end
+
+  def coaches_records_start
+    @coach_record_hash = {}
+    @games_results_per_season.each do |gr|
+      @coach_record_hash[gr['head_coach']] = {wins: 0, losses: 0, ties:0}
+    end
+    @coach_record_hash
+  end
+
+  def add_wins_losses
+    @games_results_per_season.each do |gr|
+      if gr['result'] == "WIN"
+        @coach_record_hash[gr['head_coach']][:wins] += 1
+      elsif gr['result'] == "LOSS"
+        @coach_record_hash[gr['head_coach']][:losses] += 1
+      elsif gr['result'] == "TIE"
+        @coach_record_hash[gr['head_coach']][:ties] += 1
+      end
+    end
+    @coach_record_hash
+  end
+
+  def determine_winningest_coach
+    @coach_record_hash.max_by do |coach, w_l|
+      w_l[:wins].to_f / (w_l[:wins] + w_l[:losses] + w_l[:ties])
+    end[0]
+  end
```

Comment:
I think I understand what're you're trying to accomplish with the instance variables, but I also think there is a way to do by just calling the methods that would be safer/easier to troubleshoot if something goes wrong.

I ran into a lot of issues on my war_and_peace project because I was passing around too many instance variables and was advised by an alum not to do that